packages/components/src/common/domHelper.js

```diff
@@ -0,0 +1,66 @@
+export function getPostByElement(posts, elementToFind) {
+  return posts.find(post => post.$el === elementToFind);
+}
+
+export function getElementIndexFromSiblings(targetElement) {
+  let index = 0;
+  let element = targetElement.previousElementSibling;
+  while (element !== null) {
+    element = element.previousElementSibling;
+    index += 1;
+  }
+
+  return index;
+}
+
+export function getLeftPost(el) {
+  if (!el.parentElement.previousElementSibling) return el;
+
+  const index = getElementIndexFromSiblings(el);
+
+  return el.parentElement.previousElementSibling.childNodes[index];
+}
+
+export function getRightPost(el) {
+  if (!el.parentElement.nextElementSibling) return el;
+
+  const index = getElementIndexFromSiblings(el);
+
+  return el.parentElement.nextElementSibling.childNodes[index];
+}
+
+export function getBelowPost(el) {
+  if (el.nextElementSibling === null) return el;
+
+  return el.nextElementSibling;
+}
+
+export function getAbovePost(el) {
+  if (el.previousElementSibling === null) return el;
+
+  return el.previousElementSibling;
+}
+
+export function getTopLeftMostPostEl(posts, insaneMode) {
+  const parent = posts[0].$el.parentElement;
+
+  let child = (insaneMode ? parent : parent.parentElement.firstElementChild).firstElementChild;
+  let result = getPostByElement(posts, child);
+
+  while(!result && child) {
+    child = child.nextElementSibling;
+    result = getPostByElement(posts, child);
+  }
+
+  return child;
+}
+
+export function hoverPost(selectedPost, insaneMode) {
+  const linkType = insaneMode ? 'insane__link' : 'card__link';
```

Comment:
You can add the same class for both elements to overcome the condition here.
It makes more sense to not care about the link type, in my opinion.

Reply:
yeah this is true, so silly of me. Lol